diff --git a/partmc3.py b/partmc3.py
index 24bec6b..40f4a91 100644
--- a/partmc3.py
+++ b/partmc3.py
@@ -186,13 +186,13 @@ class aero_data_t(object):
                 raise (Exception("aero_species variable not found in NetCDF file"))
             if "names" not in dir(ncf.variables["aero_species"]):
                 raise (Exception("aero_species variable does not have 'names' attribute"))
-            #self.names = ncf.variables["aero_species"].names.split(",")
+            self.names = ncf.variables["aero_species"].names.decode("utf-8").split(",")
 
             if "aero_source" not in ncf.variables.keys():
                 raise (Exception("aero_source variable not found in NetCDF file"))
             if "names" not in dir(ncf.variables["aero_source"]):
                 raise (Exception("aero_source variable does not have 'names' attribute"))
-            self.source_names = ncf.variables["aero_source"].names.split(",")
+            self.source_names = ncf.variables["aero_source"].names.decode("utf-8").split(",")
 
             for (ncf_var, self_var) in [
                 ("aero_mosaic_index", "mosaic_indices"),
@@ -764,7 +764,7 @@ class aero_particle_array_t(object):
         index_list = self.aero_data.index_list(include, exclude)
         val = numpy.zeros_like(self.raw_masses[0,:])
         for index in index_list:
-            if species_weights != None:
+            if species_weights is not None:
                 val += self.raw_masses[index,:] * species_weights[index]
             else:
                 val += self.raw_masses[index,:]
